Rspec Testing

Vagyasri opened the pull request:
**I have applied RSpec to my Enumerable Project Directory and did the following:**

- > Added specs for all the public methods I have created
- > Wrapped specs for each method in the “describe” blocks
- > Used an “it” block for each expected behavior.

Vagyasri committed:
Apply Rspec to the Directory

Create enum_spec.rb file inside spec directory

Vagyasri committed:
Set path to enumerables

Vagyasri committed:
Describe my_each method

Vagyasri committed:
Fix rubocop

Vagyasri committed:
Change example

Vagyasri committed:
Change example

Vagyasri committed:
Describe my_each_with_index method

Vagyasri committed:
Fix rubocop

Vagyasri committed:
Describe my_select

Vagyasri committed:
Add test setup to readme

Vagyasri committed:
Remove examples.txt file from spec directory

Vagyasri committed:
Describe my_all

Vagyasri committed:
Describe my_any

Vagyasri committed:
Describe my_none

Vagyasri committed:
Describe my_count

Vagyasri committed:
Describe my_map

Vagyasri committed:
Fix rubocop

Vagyasri committed:
Describe my_inject

Vagyasri committed:
Fix rubocop

Vagyasri committed:
Modify readme

Vagyasri committed:
Create rubocop

Vagyasri committed:
Fix rubocop

Vagyasri committed:
Remove gitignore

Vagyasri committed:
Create rubocop

Vagyasri committed:
Fix rubocop

george-shammar reviewed (changes requested):
### **STATUS: CHANGES REQUIRED** :heavy_check_mark: :heavy_check_mark: 

Impressive work :+1: 

**Highlights**
- All your test cases are passing.
- Your linters is not outputing any errors.
- You've tested all the enumerble methods.
Good job. :clap: 

Let's make some improvement to get this project approved.

- [ ] Your test coverage is insufficient. You have limited your test expectations to mostly if block is given and  if block is not given. The test cases  to be covered are too numerous to highlight here, but do your best to cover more cases. For example, my_each: does not mutate the original array, when self is a range, when self is a hash e.t.c. 

Increase your test coverage beyond when block is given and not given.

Implement this change and submit another CRR when you're ready.

If you need any clarification or support, drop your message here :point_down: :point_down: and ping me on slack.

**Cheers and happy coding.** :smile:

Vagyasri committed:
Add test for hash in my_each

Add test for range in my_each

Add test for hash in my_each_with_index

Add test for a range in my_each_with_index

Vagyasri committed:
Add spaces

Vagyasri committed:
Add 'when self is a hash' to my_select

Vagyasri committed:
Add 'when self is a range' to my_select

Vagyasri committed:
Give missing space

Vagyasri committed:
Add Regex to my_all
Add Class to my_all
Add Regex to my_any
Add Class to my_any
Add Regex to my_none
Add Class to my_none
Add String Array to my_inject
Add Multiplication Symbol arg to my_inject

Vagyasri committed:
Apply Rubocop

enaburekhan reviewed (approved):
# Status: Approve
**Hi Patra!
Congratulations:clap::clap:. Thank you for implementing all the changes requested by the last reviewer.
Move on with the next project. And don't forget to merge the PR and submit the project completion form.**

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.
